visualization/app/codeCharta/ui/fileChooser/fileChooser.component.spec.ts

```diff
@@ -60,8 +60,8 @@ describe("fileChooserController", () => {
 	}
 
 	describe("onImportNewFiles", () => {
-		it("should call $apply", () => {
-			fileChooserController.onImportNewFiles({ files: [] })
+		it("should call $apply and should call setNewData when data was chosen", () => {
+			fileChooserController.onImportNewFiles({ files: [{ name: "sample.cc.json" }] })
 
 			expect($rootScope.$apply).toHaveBeenCalled()
 		})
```

Comment:
Testing the FileReader here is pretty difficult and I don't see any value in this test at all. Sure, `$apply` has been called, but thats 1% of the method compared to what's actually happening. I'd just remove that test entirely.